Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -183,6 +183,16 @@ double Value::to_double() const
     return to_number().as_double();
 }
 
+size_t Value::to_size() const
+{
+    if (is_empty())
+        return 0;
+    auto number = to_number();
+    if (number.is_nan() || number.as_double() <= 0)
+        return 0;
+    return min(number.to_i32(), (i32)pow(2, 53) - 1);
```

Comment:
Does the compiler recognize `pow(2, 53)` and constant-fold it? If not, we should use a constant here instead of making a call to `pow()` every time.

Also, `to_size()` feels a bit ambiguous since `Size` is commonly used in the codebase to refer to a width+height tuple. How about we call it `to_size_t()` instead?

Reply:
> Does the compiler recognize pow(2, 53) and constant-fold it? If not, we should use a constant here instead of making a call to pow() every time.

No idea, so I'll go with a constant number :)

> Also, to_size() feels a bit ambiguous since Size is commonly used in the codebase to refer to a width+height tuple. How about we call it to_size_t() instead?

agreed!